src/player.cpp

```diff
@@ -11602,6 +11602,22 @@ std::vector<const item *> player::all_items_with_flag( const std::string &flag )
     } );
 }
 
+item player::item_with_best_of_quality( quality_id qid )
+{
+    int maxq = max_quality( qid );
+    item res_item;
+    auto items_with_quality = items_with( [qid]( const item & it ) {
+        return it.has_quality( qid );
+    } );
+    for( item *it : items_with_quality ) {
+        if( it->get_quality( qid ) == maxq ) {
+            res_item = *it;
```

Comment:
```suggestion
            return *it;
```
And add ´return item();` at the end of the function. That way you don't need to a construct a temporary item at the top of the function and you don't need to copy it.

In fact, you can even return a const item reference here. (Use `null_item_reference()` at the end of the function instead of `item()`).